src/main/java/com/hazelcast/remotecontroller/CloudManager.java

```diff
@@ -0,0 +1,428 @@
+
+        if(nullEnvVariables == "")
+            loginToCloud(baseUrl, apiKey, apiSecret);
+        else
+            throw new CloudException("Not all required environment variables are set. These are not set ones: " + nullEnvVariables);
+    }
+
+    public void loginToCloud(String url, String apiKey, String apiSecret) throws CloudException {
+        baseUrl = url;
+        bearerToken = getBearerToken(apiKey, apiSecret);
+        if(bearerToken == null)
+            throw new CloudException("Login failed");
+    }
+
+    public CloudCluster createCloudCluster(String hazelcastVersion, boolean isTlsEnabled) throws CloudException {
+        String clusterId = "";
+        try {
+            String clusterName = "test-cluster-" + UUID.randomUUID();
+            String jsonString = String.format("{" +
+                            "  \"name\": \"%s\"," +
+                            "  \"clusterTypeId\": %d," + // Serverless cluster
+                            "  \"cloudProviders\": [%d]," + // aws
+                            "  \"regions\": [%d]," + // us-west-2
+                            "  \"planName\": \"%s\"," +
+                            "  \"hazelcastVersion\": \"%s\"," +
+                            "  \"tlsEnabled\": %b" +
+                            "}",
+                    clusterName,
+                    clusterTypeId,
+                    cloudProviderId,
+                    cloudProviderRegionId,
+                    clusterPlan,
+                    hazelcastVersion,
+                    isTlsEnabled);
+
+            LOG.info(String.format("Request query: %s", jsonString));
+            try (Response response = sendPostRequest("/cluster", jsonString)){
+                ResponseBody responseBody = response.body();
+                if (responseBody == null) {
+                    throw new CloudException(String.format("Response body is null while creating a dev mode cluster: %s", response));
+                }
+                String responseString = responseBody.string();
+                handleFailedResponse(response, responseString);
+                LOG.info(maskValueOfToken(responseString));
+                clusterId = mapper.readTree(responseString).get("id").asText();
+                waitForStateOfCluster(clusterId, "RUNNING", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+                return getCloudCluster(clusterId);
+            }
+        } catch (Exception e) {
+            // Cluster id could be also empty if a problem occurred before cluster id exist
+            throw new CloudException(String.format("Create cluster with id %s is failed, Rc error is: %s", clusterId, getErrorString(e)));
+        }
+    }
+
+    private static String getErrorString(Exception e) {
+        return e + "\n" + Arrays.toString(e.getStackTrace());
+    }
+
+    private static void handleFailedResponse(Response response, String responseString) throws CloudException {
```

Comment:
Minor: the name of the method reads as the response failed, but we are checking its result here. Should we rename it?